nixos/modules/services/networking/webhook.nix

```diff
@@ -0,0 +1,189 @@
+      };
+      hooks = mkOption {
+        default = {};
+        description = "Hooks the webhook should serve";
+        example = {
+          redeploy-webhook = {
+            id = "redeploy-webhook";
+            execute-command = "/var/scripts/redeploy.sh";
+            command-working-directory = "/var/webhook";
+          };
+        };
+        type = types.attrsOf (types.submodule ({ config, ... }: {
+          options = {
+            id = mkOption {
+              type = types.str;
+              default = config._module.args.name;
+              description = "specifies the ID of your hook. This value is used to create the HTTP endpoint (http://yourserver:port/hooks/your-hook-id)";
+            };
+            execute-command = mkOption {
+              type = types.str;
+              description = "specifies the command that should be executed when the hook is triggered";
+            };
+            command-working-directory = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              description = "specifies the working directory that will be used for the script when it's executed";
+            };
+            response-message = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              description = "specifies the string that will be returned to the hook initiator";
+            };
+            response-headers = mkOption {
+              type = types.nullOr (types.listOf
+                (types.submodule {
+                  options = {
+                    name = mkOption {
+                      type = types.str;
+                      example = "X-Example-Header";
+                      description = "Header name";
+                    };
+                    value = mkOption {
+                      type = types.str;
+                      example = "it works";
+                      description = "Header value";
+                    };
+                  };
+                })
+              );
+              default = null;
+              description = "specifies the list of headers that will be returned in HTTP response for the hook";
+            };
+          };
+        }));
+      };
+      
+      success-http-response-code = mkOption {
+        type = types.nullOr types.ints.positive;
+        default = null;
+        description = "specifies the HTTP status code to be returned upon success";
+      };
+
+      
+      trigger-rule = mkOption {
+        type = types.nullOr typeTriggerRule;
+        default = null;
+        description = "specifies the list of headers that will be returned in HTTP response for the hook";
+      };
+
+      # TODO: add other options from https://github.com/adnanh/webhook/blob/master/docs/Hook-Definition.md
+    };
+  };
+  
+  imports = [];
+  
+  config = mkIf cfg.enable {
+
+    systemd.packages = [ pkgs.webhook ];
```

Comment:
This shouldn't be necessary, because you already reference it directly in the start script (where it can trigger a restart when necessary, which is good)